pkg/core/mem_pool.go

```diff
@@ -0,0 +1,250 @@
+package core
+
+import (
+	"sort"
+	"time"
+
+	"github.com/CityOfZion/neo-go/pkg/core/transaction"
+	"github.com/CityOfZion/neo-go/pkg/util"
+)
+
+// PoolItem represents a transaction in the the Memory pool.
+type PoolItem struct {
+	txn       *transaction.Transaction
+	timeStamp time.Time
+	fee       Feer
+}
+
+// PoolItems slice of PoolItem
+type PoolItems []*PoolItem
+
+// MemPool stores the unconfirms transactions.
+type MemPool struct {
+	unsortedTxn                 map[util.Uint256]*PoolItem
+	unverifiedTxn               map[util.Uint256]*PoolItem
+	sortedHighPrioTxn           PoolItems
+	sortedLowPrioTxn            PoolItems
+	unverifiedSortedHighPrioTxn PoolItems
+	unverifiedSortedLowPrioTxn  PoolItems
+
+	capacity int
+}
+
+func (p PoolItems) Len() int           { return len(p) }
+func (p PoolItems) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+func (p PoolItems) Less(i, j int) bool { return (p[i].CompareTo(p[j]) < 0) }
+
+// CompareTo returns the difference between two PoolItems.
+// difference < 0 implies p < otherP.
+// difference = 0 implies p = otherP.
+// difference > 0 implies p > otherP.
+func (p *PoolItem) CompareTo(otherP *PoolItem) int {
+	if otherP == nil {
+		return 1
+	}
+
+	if p.fee.IsLowPriority(p.txn) && p.fee.IsLowPriority(otherP.txn) {
+		thisIsClaimTx := p.txn.Type == transaction.ClaimType
+		otherIsClaimTx := otherP.txn.Type == transaction.ClaimType
+
+		if thisIsClaimTx != otherIsClaimTx {
+			// This is a claim Tx and other isn't.
+			if thisIsClaimTx {
+				return 1
+			}
+			// The other is claim Tx and this isn't.
+			return -1
+		}
+	}
+
+	// Fees sorted ascending
+	pFPB := p.fee.FeePerByte(p.txn)
+	otherFPB := p.fee.FeePerByte(otherP.txn)
+	if ret := pFPB.CompareTo(otherFPB); ret != 0 {
+		return ret
+	}
+
+	pNF := p.fee.NetworkFee(p.txn)
+	otherNF := p.fee.NetworkFee(otherP.txn)
+	if ret := pNF.CompareTo(otherNF); ret != 0 {
+		return ret
+	}
+
+	// Transaction hash sorted descending
+	return otherP.txn.Hash().CompareTo(p.txn.Hash())
+}
+
+// Count returns the total number of uncofirm transactions.
+func (mp MemPool) Count() int {
+	return len(mp.unsortedTxn) + len(mp.unverifiedTxn)
+}
+
+// ContainsKey checks if a transactions hash is in the MemPool.
+func (mp MemPool) ContainsKey(hash util.Uint256) bool {
+	if _, ok := mp.unsortedTxn[hash]; ok {
+		return true
+	}
+
+	if _, ok := mp.unverifiedTxn[hash]; ok {
+		return true
+	}
+
+	return false
+}
+
+// TryAdd try to add the PoolItem to the MemPool.
+func (mp MemPool) TryAdd(hash util.Uint256, pItem *PoolItem) bool {
+	var pool PoolItems
+
+	if _, ok := mp.unsortedTxn[hash]; ok {
+		return false
+	}
+	mp.unsortedTxn[hash] = pItem
+
+	if pItem.fee.IsLowPriority(pItem.txn) {
+		pool = mp.sortedLowPrioTxn
+	} else {
+		pool = mp.sortedHighPrioTxn
+	}
+	pool = append(pool, pItem)
+	sort.Sort(pool)
+
+	if mp.Count() > mp.capacity {
+
+	}
```

Comment:
empty statement?

Reply:
I will add the missing code. :sweat_smile: